### PR TITLE
Added variant annotation subworkflow with snpeff and VEP

### PR DESCRIPTION
**Pull request by praveenraj2018**
Added variant annotation sub-workflow with snpeff and VEP

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/rnavar/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/rnavar _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: .nf-core.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+lint:
+  files_unchanged:
+    - .github/workflows/linting.yml
+    - assets/multiqc_config.yaml
+    - assets/nf-core-rnavar_logo.png
+    - docs/images/nf-core-rnavar_logo.png
```
Comment (maxulysse) [inline .nf-core.yml:6]: You should not need these two, I believe we have a regular nf-core logo for rnavar

File: conf/test.config [hunk 2]
```diff
@@ -23,10 +23,12 @@ params {
     input = 'assets/samplesheet_test.csv'
 
     // Genome references
+    genome              = 'GRCh38'
```
Comment (maxulysse) [inline conf/test.config:26]: I'd recommend to test not with human as the db to download is huge.
Can you try if it annotates with C.elegans (even if variants are human).
It's what I do in Sarek

File: conf/test.config [hunk 3]
```diff
@@ -36,4 +38,11 @@ params {
 
     // STAR index (optional)
     star_index          = 'https://github.com/nf-core/test-datasets/raw/modules/data/genomics/homo_sapiens/genome/index/star/star.tar.gz'
+
+    // Annotation
+    snpeff_db           = 'GRCh38.99'
+    vep_genome          = 'GRCh38.p13'
+    vep_cache_version   = '104'
+    vep_species         = 'homo_sapiens'
```
Comment (maxulysse) [inline conf/test.config:46]: I'd recommend to test not with human as the db to download is huge.
Can you try if it annotates with C.elegans (even if variants are human).
It's what I do in Sarek

Comment (praveenraj2018) [reply conf/test.config:46]: oh ok. I saw that in Sarek. I can do that

File: .nf-core.yml (hunk 1 above)
Comment (praveenraj2018) [reply .nf-core.yml:6]: not sure which logo are you referring to. I haven't changed these images...

File: conf/test.config (hunk 3 above)
Comment (praveenraj2018) [reply conf/test.config:46]: Added C.elegans!

**Comment by maxulysse** [comment]
Can you try to fix the nf-core lint?
if `nf-core lint --fix files_unchanged` is not working properly, just removed the files that are still listed an an error.

File: .nf-core.yml (hunk 1 above)
Comment (maxulysse) [reply .nf-core.yml:6]: Then, why are they here in the files_unchanged?

**Review by maxulysse — APPROVED** [review]
LGTM

**Comment by maxulysse** [comment]
Maybe we can add https://github.com/nf-core/configs/blob/master/conf/pipeline/sarek/munin.config#L18-L20 to https://github.com/nf-core/configs/blob/master/conf/munin.config so that we can use the annotation cache in munin already in rnavar

**Pull request merged**